src/runtime/components/image.mixin.ts

```diff
@@ -25,41 +30,52 @@ export const imageMixin = {
     usemap: { type: String, default: undefined },
     longdesc: { type: String, default: undefined },
     ismap: { type: Boolean, default: undefined },
-    crossorigin: { type: Boolean, default: undefined },
+    crossorigin: { type: [Boolean, String] as unknown as () => boolean | '' | 'anonymous' | 'use-credentials', default: undefined, validator: val => ['anonymous', 'use-credentials', '', true, false].includes(val) },
     loading: { type: String, default: undefined },
-    decoding: { type: String, default: undefined }
+    decoding: { type: String as () => 'async' | 'auto' | 'sync', default: undefined, validator: val => ['async', 'auto', 'sync'].includes(val) }
   },
   computed: {
-    nImgAttrs () {
+    nImgAttrs (): {
+      width?: number
+      height?: number
+      alt?: string
+      referrerpolicy?: string
+      usemap?: string
+      longdesc?: string
+      ismap?: boolean
+      crossorigin?: '' | 'anonymous' | 'use-credentials'
+      loading?: string
+      decoding?: 'async' | 'auto' | 'sync'
+      } {
       return {
-        width: this.width,
-        height: this.height,
+        width: parseSize(this.width),
+        height: parseSize(this.height),
         alt: this.alt,
         referrerpolicy: this.referrerpolicy,
         usemap: this.usemap,
         longdesc: this.longdesc,
         ismap: this.ismap,
-        crossorigin: this.crossorigin,
+        crossorigin: this.crossorigin === true ? 'anonymous' : this.crossorigin || undefined,
         loading: this.loading,
         decoding: this.decoding
       }
     },
-    nModifiers () {
+    nModifiers (): { width?: number, height?: number, format?: string, quality?: string | number, background?: string, fit?: string } & Record<string, any> {
```

Comment:
This return type (and rest of the functions) are already inferred by typescript. (also mkdist at least currently for some reason is not preserving types at all)